File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Python/s967693306_commented.py
import sys

def main():
    parts = sys.stdin.readline().split()
    n = int(parts[0])
    m = int(parts[1])
    conditions = {}
    conflict = False
    for _ in range(m):
        parts = sys.stdin.readline().split()
        s = int(parts[0])
        c = int(parts[1])
        if s in conditions:
            if conditions[s] != c:
                print(-1)
                return
        conditions[s] = c
    for i in range(1000):
        s = str(i)
        if len(s) < n:
            continue
        flag = True
        for j in range(n):
            current_digit = int(s[j])
            pos = j + 1
            if pos in conditions:
                if current_digit != conditions[pos]:
                    flag = False
                    break
            else:
                if current_digit != 0:
                    if not (j == 0 and current_digit == 1):
                        flag = False
                        break
        if flag and len(s) == n:
            print(s)
            return
    print(-1)

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s967693306_commented.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	var n, m int
	fmt.Scan(&n, &m)

	conditions := make(map[int]int)

	for i := 0; i < m; i++ {
		var s, c int
		fmt.Scan(&s, &c)
		if existing, ok := conditions[s]; ok {
			if existing != c {
				fmt.Println(-1)
				return
			}
		}
		conditions[s] = c
	}

	for i := 0; i < 1000; i++ {
		str := strconv.Itoa(i)
		if len(str) != n {
			continue
		}

		valid := true
		for j := 0; j < n; j++ {
			pos := j + 1
			required, exists := conditions[pos]
			currentDigit := int(str[j] - '0')

			if exists {
				if currentDigit != required {
					valid = false
					break
				}
			} else {
				if currentDigit != 0 {
					if j == 0 && currentDigit == 1 {
						continue
					} else {
						valid = false
						break
					}
				}
			}
		}

		if valid {
			fmt.Println(str)
			return
		}
	}

	fmt.Println(-1)
}

// End of Code
